### PR TITLE
Refactor/fe=fn 25 cartscreen

### PR DESCRIPTION
**Pull request by Darshan6069**
update cart Screen

### CODE REVIEW

**Review by Harshyam — APPROVED** [review]

File: lib/Domain/categories_model/categories_model.dart [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class CategoriesModel {
+  final String name;
+  final String iconImage;
+
+  CategoriesModel({required this.name, required this.iconImage});
```
Comment (nikkielevancex) [inline lib/Domain/categories_model/categories_model.dart]: suggestion: use freezed

File: lib/Presentation/Authentication/login_screen.dart [hunk 2]
```diff
@@ -75,8 +76,8 @@ class _LoginScreenState extends State<LoginScreen> {
                 ),
               ),
             ),
-            const ButtonWidget(
-              buttonName: AppStrings.signIn,
+             const ButtonWidget(
+              buttonName: AppStrings.signIn, navigatorScreenName: HomeScreen(),
```
Comment (nikkielevancex) [inline lib/Presentation/Authentication/login_screen.dart:80]: **question**: why are you navigating to HomeScreen on tap of signIn.
If its kept for temporary then add a TODO

File: lib/Presentation/tabs_screens/screens/home_screen/search_screen/search_screen.dart [hunk 3]
```diff
@@ -1,74 +1,103 @@
...
   @override
   Widget build(BuildContext context) {
     return Scaffold(
-      appBar: AppBar(
-        title: const Text('Search Food'),
-        centerTitle: true,
-        backgroundColor: Colors.white,
-        elevation: 0,
+      resizeToAvoidBottomInset: false,
+      appBar: const CommonAppBar(
```
Comment (nikkielevancex) [inline lib/Presentation/tabs_screens/screens/home_screen/search_screen/search_screen.dart]: praise: good work !

File: lib/Presentation/tabs_screens/screens/home_screen/search_screen/search_screen.dart [hunk 4]
```diff
@@ -1,74 +1,103 @@
...
-                prefixIcon: const Icon(Icons.search),
-                suffixIcon: const Icon(Icons.filter_alt_outlined),
-                border: OutlineInputBorder(
-                  borderRadius: BorderRadius.circular(12),
-                ),
-              ),
+            TextformfieldWidget(
+              controller: searchController,
+              hint: AppStrings.searchFood,
+              priFixIcon: Iconses.search,
```
Comment (nikkielevancex) [inline lib/Presentation/tabs_screens/screens/home_screen/search_screen/search_screen.dart]: **suggestion**: keep the class name as AppIcons

File: lib/Presentation/tabs_screens/screens/home_screen/search_screen/search_widget/recent_order.dart [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(vertical: 8.0),
+      child: Row(
+        children: [
+          Container(
+            width: 70,
+            height: 70,
```
Comment (nikkielevancex) [inline lib/Presentation/tabs_screens/screens/home_screen/search_screen/search_widget/recent_order.dart]: **suggestion**: avoid using static values

File: lib/Presentation/tabs_screens/screens/home_screen/search_screen/search_widget/recent_order.dart [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+            decoration: BoxDecoration(
+              borderRadius: BorderRadius.circular(8),
+              image: const DecorationImage(
+                image: AssetImage('assets/profile_screen/burger.png'),
+                // Example image, replace with actual asset
```
Comment (nikkielevancex) [inline lib/Presentation/tabs_screens/screens/home_screen/search_screen/search_widget/recent_order.dart]: add a TODO comment

File: lib/Presentation/tabs_screens/screens/home_screen/search_screen/search_widget/recent_search_data.dart [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+  final String recentSearch;
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(vertical: 4.0),
+      child: Row(
+        children: [
+          const Icon(Icons.search, color: Colors.grey),
```
Comment (nikkielevancex) [inline lib/Presentation/tabs_screens/screens/home_screen/search_screen/search_widget/recent_search_data.dart]: do not use color directly from Colors library

File: lib/Presentation/tabs_screens/screens/profile_screen/settings_screen/language_selector.dart [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final List<Map<String, String>> languages = [
+      {'name': 'Indonesia', 'flag': '🇮🇩'},
+      {'name': 'English (US)', 'flag': '🇺🇸'},
+      {'name': 'Thailand', 'flag': '🇹🇭'},
+      {'name': 'Chinese', 'flag': '🇨🇳'},
+    ];
```
Comment (nikkielevancex) [inline lib/Presentation/tabs_screens/screens/profile_screen/settings_screen/language_selector.dart]: **suggestion**: keep this list in AppConstants file

File: lib/Presentation/tabs_screens/screens/profile_screen/settings_screen/language_selector.dart [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+
+    return Container(
+      decoration: const BoxDecoration(
+        color: Colors.white,
```
Comment (nikkielevancex) [inline lib/Presentation/tabs_screens/screens/profile_screen/settings_screen/language_selector.dart]: do not use color directly from Colors library

File: lib/core/constant/strings.dart [hunk 10]
```diff
@@ -33,9 +33,19 @@ class AppStrings{
...
 
   // images path
+  static const burgerIcon= 'assets/search_screen_image/burger.png';
+  static const tacoIcon= 'assets/search_screen_image/taco.png';
+  static const drinkIcon= 'assets/search_screen_image/drink.png';
+  static const pizzaIcon= 'assets/search_screen_image/pizza.png';
 
   static const googleIcon = 'assets/AuthImages/google.svg';
   static const facebookIcon = 'assets/AuthImages/facebook.svg';
   static const appleIcon = 'assets/AuthImages/apple.svg';
```
Comment (nikkielevancex) [inline lib/core/constant/strings.dart]: **suggestion**: create a new class called AssetImages and keep this in that file

**Review by nikkielevancex — APPROVED** [review]

**Pull request merged**